src/deluge/model/song/song.h

```diff
@@ -346,3 +350,26 @@ class Song final : public TimelineCounter {
 extern Song* currentSong;
 extern Song* preLoadedSong;
 extern int8_t defaultAudioClipOverdubOutputCloning;
+
+[[gnu::always_inline]] extern inline Clip* getCurrentClip() {
+	return currentSong->currentClip;
+};
+
+[[gnu::always_inline]] extern inline InstrumentClip* getCurrentInstrumentClip() {
+	return (InstrumentClip*)currentSong->currentClip;
```

Comment:
These aren't safe, no guarantee that's an instrument clip before you cast it

Reply:
How do we make them safe? They were declared this way in the other classes. Was it safe because in those classes you had a guarantee you were dealing with an instrument clip?